Bring back the RealitySystemSupport SoftLink

etiennesegonzac opened the pull request:
#### 2e65fd05d66094570c85625c5f31e99cedb11612
<pre>
Bring back the RealitySystemSupport SoftLink
<a href="https://bugs.webkit.org/show_bug.cgi?id=259610">https://bugs.webkit.org/show_bug.cgi?id=259610</a>
&lt;rdar://113038445&gt;

Reviewed by Tim Horton.

Keeping the RealitySystemSupport link soft and optional for now.

* Source/WebKit/Configurations/WebKit.xcconfig:
Remove the linker flags.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/Platform/spi/visionos/RealitySystemSupportSPI.h: Removed.
Remove the SPI header.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
(WebKit::interactionRegionLayerClass):
(WebKit::interactionRegionEffectUserInfo):
(WebKit::configureLayerForInteractionRegion):
Use the re-introduced optional soft link.

Canonical link: <a href="https://commits.webkit.org/266404@main">https://commits.webkit.org/266404@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d7eb4c0b3b80d8ece04836ee70b866dd3569916b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13732 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14046 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14379 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15468 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13041 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16554 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14127 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15719 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13899 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14518 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11624 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16167 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11807 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19421 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12882 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12547 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15764 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13077 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10953 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12345 "Built successfully") | | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3342 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16678 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12919 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
```diff
@@ -32,7 +32,13 @@
 #import "RemoteLayerTreeHost.h"
 
 #if PLATFORM(VISION)
-#import "RealitySystemSupportSPI.h"
```
You certainly don't have to delete the SPI header, it's orthogonal to the soft linking. But this is fine too (as a mostly-revert).

etiennesegonzac committed:
Bring back the RealitySystemSupport SoftLink
https://bugs.webkit.org/show_bug.cgi?id=259610
<rdar://113038445>

Reviewed by Tim Horton.

Keeping the RealitySystemSupport link soft and optional for now.

* Source/WebKit/Configurations/WebKit.xcconfig:
Remove the linker flags.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/Platform/spi/visionos/RealitySystemSupportSPI.h: Removed.
Remove the SPI header.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeInteractionRegionLayers.mm:
(WebKit::interactionRegionLayerClass):
(WebKit::interactionRegionEffectUserInfo):
(WebKit::configureLayerForInteractionRegion):
Use the re-introduced optional soft link.

Canonical link: https://commits.webkit.org/266404@main
